.github/workflows/dotnet-desktop.yml

```diff
@@ -0,0 +1,56 @@
+name: .Net Core Desktop
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "*" ]
+
+jobs:
+  build-test:
+    name: Build and Test
+    runs-on: windows-latest
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    
+    - name: Install .NET Core
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: 8.0
+
+    - name: Setup MSBuild.exe
+      uses: microsoft/setup-msbuild@v2
+
+    - name: Install ntools and dev tools
+      run: |
+        cd ./DevSetup
+        ./DevSetup.ps1
+      shell: pwsh
+      working-directory: ${{ github.workspace }}
+
+    - name: Set ProgramFiles Path
+      run: |
+        $programFilesPath = [System.Environment]::GetFolderPath('ProgramFiles')
+        echo "ProgramFilesPath=$programFilesPath" | Out-File -FilePath $env:GITHUB_ENV -Append
+      shell: pwsh
+      working-directory: ${{ github.workspace }}
+   
+    - name: Configure Git
+      run: |
+        git config --global user.name "gitHub-actions"
+        git config --global user.email "actions@github.com"
+
+    - name: Build using ntools
+      run: |
+        & "$env:ProgramFilesPath/nbuild/nb.exe" solution
```

Comment:
@ahmdhjj I recommend that we can combine `solution` and `test` target and use `staging` target,

Reply:
@naz-hage done